Account Management

rxu183 opened the pull request:
Adds settings button under the avatar. Allows a user to change their username, password, and delete their account by interacting with the basemodel on the backend.

gyuheon0h commented:
If frontend was changed, would be nice to include video or screenshots showing the change

gyuheon0h commented on server/src/routes/settings/settings.ts:
```diff
@@ -0,0 +1,121 @@
+
+// Original route: /api/update-profile
+// New route: /api/settings/update-profile
+settingsRouter.post('/update-profile', async (req: AuthenticatedRequest, res) => {
```
if this is authenticated request, we should pass in the `authorize` middleware. This lets us get userId from the request also

rxu183 replied:
fixed!

gyuheon0h commented on server/src/routes/settings/settings.ts:
```diff
@@ -0,0 +1,121 @@
+
+// Original route: /api/delete-account
+// New route: /api/settings/delete-account
+settingsRouter.delete('/delete-account', async (req: AuthenticatedRequest, res) => {
```
Same as above

rxu183 replied:
fixed!

rxu183 commented:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/c7267586-acd0-412c-ab05-82372a29bfd3" />

Changed authorization, and changed settings page into a dialog.

itszapenguin reviewed (approved):
wagmi
